cmd/state-installer/cmd.go

```diff
@@ -354,20 +375,43 @@ func postInstallEvents(out output.Outputer, an analytics.Dispatcher, params *Par
 		an.Event(AnalyticsFunnelCat, "forward-activate-default")
 
 		out.Print(fmt.Sprintf("\nRunning `[ACTIONABLE]state activate --default %s[/RESET]`\n", params.activateDefault.String()))
-		if _, _, err := exeutils.ExecuteAndPipeStd(stateExe, []string{"activate", params.activateDefault.String(), "--default"}, env); err != nil {
+		if _, _, err := exeutils.ExecuteAndPipeStd(stateExe, []string{"activate", params.activateDefault.String(), "--default"}, envSlice(binPath)); err != nil {
 			an.EventWithLabel(AnalyticsFunnelCat, "forward-activate-default-err", err.Error())
 			return errs.Wrap(err, "Could not activate %s, error returned: %s", params.activateDefault.String(), errs.JoinMessage(err))
 		}
+	default:
+		if !isUpdate {
+			ss := subshell.New(cfg)
+			ss.SetEnv(envMap(binPath))
+			if err := ss.Activate(nil, cfg, out); err != nil {
+				return errs.Wrap(err, "Subshell setup; error returned: %s", errs.JoinMessage(err))
+			}
+			ers := ss.Errors()
+			for err = range ers {
+				return errs.Wrap(err, "Subshell execution; error returned: %s", errs.JoinMessage(err))
```

Comment:
Return in a for loop?

Reply:
I've updated this to handle the channel as it is in the activate runner. I expect subshell modifications will result in a cleaner API so that a channel is not directly exposed. Otherwise, if that is preferred, I'd like to file a ticket to create a helper function in the errs package.